### PR TITLE
Zustand feature

### PR DESCRIPTION
**Pull request by Hugovs2000**
Made use of Zustand to implement a continue reading function that takes the user to their last read manga chapter.
Also implemented a favourites feature where a user can favourite a manga on the manga's detail page and also view their favourite mangas by navigating to the favourites page from the hamburger menu. Slightly fixed some code here and there as well.

### CODE REVIEW

**Review by Julianna-Venter — APPROVED** [review]
Looks good to me, just carefully double check behaviour, especially with your useEffect and how data is being updated. I'd just spend some time trying to break it, just to make sure state is staying consistent. But I haven't seen any obvious weird behaviours, so well done!

**Review by tanadendere — APPROVED** [review]
This is looking good, well done!

File: src/routes/__root.tsx [hunk 1]
```diff
@@ -25,11 +29,27 @@ function RootComponent() {
         <input id="my-drawer" type="checkbox" className="drawer-toggle" />
         <div className="drawer-content flex flex-col min-h-full">
           <Navbar className="fixed z-50 navbar bg-zinc-900 flex flex-shrink-0" />
-          <div className="pt-16 flex-grow">
+          <div className="pt-16 flex-grow relative">
             <ScrollRestoration />
             <Outlet />
+            {!isHidden &&
+              lastReadPage?.[0].length > 0 &&
```
Comment (Musa-Mabasa) [inline src/routes/__root.tsx:36]: Safety check here: `lastReadPage?.[0]?.length`

File: src/modules/SeeMorePage/components/SeeMoreSeasonal.tsx [hunk 2]
```diff
@@ -10,30 +10,35 @@ export default function SeeMoreSeasonal({
...
-                b2key={item?.md_covers?.[0].b2key!}
-                title={item.title!}
-              />
-            </Link>
-          ))}
+          .map(
+            (item) =>
+              item.slug &&
+              item.title &&
+              item?.md_covers?.[0].b2key && (
```
Comment (Musa-Mabasa) [inline src/modules/SeeMorePage/components/SeeMoreSeasonal.tsx:27]: Safety check here as well.

**Review by Musa-Mabasa — CHANGES_REQUESTED** [review]
Everything looks good, just a few missing safety checks from my comments

File: src/routes/__root.tsx (hunk 1 above)
Comment (Hugovs2000) [reply src/routes/__root.tsx:36]: Thank you 🙏🏻

File: src/modules/SeeMorePage/components/SeeMoreSeasonal.tsx (hunk 2 above)
Comment (Hugovs2000) [reply src/modules/SeeMorePage/components/SeeMoreSeasonal.tsx:27]: Thank you 🙏🏻

File: src/modules/ReadMangaPage/components/TopInfoBar.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default function TopInfoBar({
   chapterData: ChapterDetails;
 }) {
   return (
-    <div className="h-10 fixed top-16 p-2 w-full flex items-center bg-emerald-700 rounded-b-md">
+    <div className="h-10 fixed top-16 p-2 w-full flex items-center bg-emerald-700 rounded-b-md z-20">
```
Comment (LesediMoko) [inline src/modules/ReadMangaPage/components/TopInfoBar.tsx:13]: would z-10 not be sufficient here?

File: src/state/state-service.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { create } from "zustand";
+import { createJSONStorage, persist } from "zustand/middleware";
+import MangaState from "../models/StoreState";
+
+export const useMangaStore = create<MangaState>()(
+  persist(
```
Comment (LesediMoko) [inline src/state/state-service.ts:6]: Great use of persistence to keep the track of the LastRead and favourites 👏

File: src/modules/MangaDetailsPage/components/MangaHeader.tsx [hunk 5]
```diff
@@ -1,9 +1,34 @@
...
+  const existingFavs = useMangaStore((state) => state.favourites);
+
+  let isFav: boolean;
+
+  topData?.comic?.slug && existingFavs.includes(topData?.comic?.slug)
+    ? (isFav = true)
+    : (isFav = false);
+
+  const handleFavClick = () => {
+    if (topData?.comic?.slug) {
```
Comment (LesediMoko) [inline src/modules/MangaDetailsPage/components/MangaHeader.tsx:23]: I see that the logic should only evaluate if topData.comic.slug exists (without an else block). Maybe consider using a short circuit. i.e. ` topData?.comic?.slug && {//second if logic) `

**Review by LesediMoko — APPROVED** [review]
You put in a lot of good, hard work in this one. Super impressive 👏. Just minor comments regarding z-indices and conditional statements but all-in-all, amazing work Hugo 🏄🏽‍♂️

**Review by Musa-Mabasa — APPROVED** [review]
Everything looks good 👍

File: src/modules/MangaDetailsPage/components/MangaHeader.tsx (hunk 5 above)
Comment (Hugovs2000) [reply src/modules/MangaDetailsPage/components/MangaHeader.tsx:23]: Thank you 👍🏻

File: src/modules/ReadMangaPage/components/TopInfoBar.tsx (hunk 3 above)
Comment (Hugovs2000) [reply src/modules/ReadMangaPage/components/TopInfoBar.tsx:13]: I tried it, the chapter images scroll over the top info bar, only works with z-20. Thanks though

**Pull request merged**